src/MetamaskInpageProvider.js

```diff
@@ -22,12 +22,29 @@ const {
 
 module.exports = class MetamaskInpageProvider extends SafeEventEmitter {
 
-  constructor (connectionStream, shouldSendMetadata = true) {
+  /**
+   * @param {Object} connectionStream - A Node.js stream
+   * @param {Object} opts - An options bag
+   * @param {number} opts.maxEventListeners - The maximum number of event listeners
+   * @param {boolean} opts.shouldSendMetadata - Whether the provider should send page metadata
+   */
+  constructor (
+    connectionStream,
+    { shouldSendMetadata = true, maxEventListeners = 100 },
+  ) {
+
+    if (
+      typeof shouldSendMetadata !== 'boolean' || typeof maxEventListeners !== 'number'
+    ) {
+      throw new Error('Invalid options.')
+    }
 
     super()
 
     this.isMetaMask = true
 
+    this.setMaxEventListeners(maxEventListeners)
```

Comment:
I think this function is misnamed (https://nodejs.org/docs/latest-v10.x/api/events.html#events_emitter_setmaxlisteners_n)

```suggestion
    this.setMaxListeners(maxEventListeners)
```